cmd/save.go

```diff
@@ -65,13 +65,14 @@ commit message and title to the save.`,
 	cmd.Flags().StringVarP(&o.Message, "message", "m", "", "commit message for save")
 	cmd.Flags().StringVarP(&o.BodyPath, "body", "", "", "path to file or url of data to add as dataset contents")
 	cmd.MarkFlagFilename("body")
-	cmd.Flags().StringVarP(&o.Recall, "recall", "", "", "restore revisions from dataset history")
 	// cmd.Flags().BoolVarP(&o.ShowValidation, "show-validation", "s", false, "display a list of validation errors upon adding")
+	cmd.Flags().BoolVar(&o.Apply, "apply", false, "apply a transformation and save the result")
+	cmd.Flags().BoolVar(&o.NoApply, "no-apply", false, "don't apply any transforms that are added")
```

Comment:
this flag feels a little clunky, but playing with this branch in the wild makes sense. Forcing the user to explicitly say what they want to do with the transform is good, but the ergonomics of these two flags _and_ the needed `--file` flag feels like it could use improvement. How do you feel about making `--apply` and `--no-apply` accept path args that must be transforms?

```
$ qri save --apply tf.star me/dataset
$ qri save --no-apply tf.star me/dataset
```

On one hand it violates the multiple `--file` flag convention, but we're already doing that with `--body`. 🤷

Reply:
I've gone back and forth on the need for a `--file` flag. Looking at the [RFC](https://github.com/qri-io/rfcs/blob/master/text/0027-transform_application.md), which also explains that exact rational for `--no-apply`, there is also no `--file` flag. The reason I thought it should be put in place is the ergonomics of:

1) Crafting an apply command to develop a transform.
2) Running the transform iteratively and eventually finishing it.
3) Changing it to make a commit by simply insert the string `save --` between `qri` and `apply`.

Example:
```
qri apply --file dev_transform.star me/my_dataset
```
becomes:
```
qri save --apply --file dev_transform.star me/my_dataset
```

Removing the need for `--file` for `apply` would break that 3rd step, since it would turn a command with 2 positions arguments into a command with only 1. However, I agree that requiring `--file` makes step 1 a bit more awkward. Looking to see if you have any thoughts on this specifically because I'm rather split-minded.

Reply:
I'm with you on the fence. Ok, let's keep the PR as-written then